spec.md

```diff
@@ -2606,6 +2606,24 @@ enumerate(["zero", "one", "two"])               # [(0, "zero"), (1, "one"), (2,
 enumerate(["one", "two"], 1)                    # [(1, "one"), (2, "two")]
 ```
 
+### fail
+
+The `fail(*args)` function causes execution to fail
+with an error message that includes the string forms of the argument values.
```

Comment:
Explicitly mention that the form is not necessarily the same as what is returned by `str` or `repr`. (Optionally also explain that this is an intentional design choice to allow for information to be emitted by fail without necessarily exposing it to programmatic access by the Starlark code, e.g. non-deterministic information in deterministic applications like Bazel.)

Reply:
In the interests of reducing non-normative text in the spec, I added this:

"""
...
The precise formatting depends on the implementation.                                                

<!--                                                                                                 
Note:                                                                                                
                                                                                                     
Neither the template of the error message nor the formatting of the                                  
values is prescribed here. Implementations may use a richer representation                           
than str or repr, with additional debugging information.                                             
The error message is not observable by Starlark programs.                                            
-->
"""

Reply:
Perfect.